llm/openai/openai.go

```diff
@@ -219,27 +253,29 @@ func (o *OpenAI) stream(
 		return fmt.Errorf("%w: %w", ErrOpenAIChat, err)
 	}
 
-	var messages []*thread.Message
 	var content string
+	var messages []*thread.Message
+	var allToolCalls []openai.ToolCall
+	var currentToolCall openai.ToolCall
 	for {
 		response, errRecv := stream.Recv()
 		if errors.Is(errRecv, io.EOF) {
-			o.streamCallbackFn(EOS)
-
-			if len(content) > 0 {
-				messages = append(messages, thread.NewAssistantMessage().AddContent(
-					thread.NewTextContent(content),
-				))
-			}
+			messages = o.processErrorResponse(messages, content, &currentToolCall, allToolCalls)
```

Comment:
use the new function name

```suggestion
			messages = o.handleEndOfStream(messages, content, &currentToolCall, allToolCalls)
```